lib/mix/lib/mix/scm/git.ex

```diff
@@ -282,6 +321,8 @@ defmodule Mix.SCM.Git do
   end
 
   defp default_branch() do
+    # Note: the `set-head -a` command requires the remote reference to be
+    # fetched first.
```

Comment:
Added this note because the most obvious code would be to "compute" `rev` in `checkout/2` in a single expression as it used to be, but that doesn't work.